runtime/gc_base/GCExtensions.cpp

```diff
@@ -267,3 +273,71 @@ MM_GCExtensions::computeDefaultMaxHeap(MM_EnvironmentBase *env)
 
 	memoryMax = MM_Math::roundToFloor(heapAlignment, memoryMax);
 }
+
+#if defined(J9VM_GC_ENABLE_DOUBLE_MAP)
+#if !defined(LINUX) || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF)
+/* Double map is only supported on LINUX 64 bit Systems for now */
+#error "Platform not supported by Double Map API"
+#endif /* !LINUX || !defined(J9VM_ENV_DATA64) || defined(OMRZTPF) */
+void* 
+MM_GCExtensions::doubleMapArraylets(MM_EnvironmentBase* env, J9Object *objectPtr) 
+{
+	J9JavaVM *javaVM = getJavaVM();
+	PORT_ACCESS_FROM_ENVIRONMENT(env);
+
+	GC_ArrayletLeafIterator arrayletLeafIterator(javaVM, (J9IndexableObject*)objectPtr);
+	UDATA arrayletLeafCount = arrayletLeafIterator.getNumLeafs();
+	MM_Heap *heap = getHeap();
+
+	void *result = NULL;
+	void **arrayletLeaveAddrs = (void **)j9mem_allocate_memory(arrayletLeafCount * sizeof(uintptr_t), J9MEM_CATEGORY_VM);
```

Comment:
Usually in GC code we are using `MM_Forge *forge` (can be get from `MM_GCExtensionsBase` or `MM_EnvironmentBase`) for allocate/free: `forge->allocate()` and `forge->free()`. You can use `AllocationCategory::GC_HEAP`. 
There is an example of allocation request:
```
	_extensions->referenceObjectLists = (MM_ReferenceObjectList *)env->getForge()->allocate((sizeof(MM_ReferenceObjectList) * listCount), MM_AllocationCategory::FIXED, J9_GET_CALLSITE());
```